### ISSUE TITLE
micromamba build error of alphafold3 on Linux with python3.11.

### ISSUE THREAD

**Issue by ZhihaoXie** [google-deepmind/alphafold3#223]
I've found a build error of alphafold3 on Linux with python3.11.
`
micromamba create -n af3 python=3.11
micromamba activate af3

git clone https://github.com/google-deepmind/alphafold3.git
cd alphafold3
conda install -c bioconda hmmer
pip3 install -r dev-requirements.txt
pip3 install --no-deps .
`

After this, a build error occurred:

`
  /tmp/tmp2fz4ptpk/build/_deps/cifpp-src/include/cif++/utilities.hpp:193:2: error: ‘requires’ does not name a type
    193 |  requires std::is_assignable_v<std::string_view, T>
        |  ^~~~~~~~
  [200/235] Building CXX object _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o
  FAILED: _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o
  /usr/bin/g++  -pthread -B /data/xiezhihao/software/micromamba/envs/af3/compiler_compat -DBOOST_REGEX_STANDALONE=1-DCACHE_DIR=\"/tmp/tmp2fz4ptpk/wheel/platlib/var/cache/libcifpp\" -DDATA_DIR=\"/tmp/tmp2fz4ptpk/wheel/platlib/share/libcifpp\" -DUSE_BOOST_REGEX=1 -I/tmp/tmp2fz4ptpk/build/_deps/cifpp-src/include -I/tmp/tmp2fz4ptpk/build/_deps/boost-rx-src/include -I/tmp/tmp2fz4ptpk/build/_deps/my-eigen3-src -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -O3 -DNDEBUG -std=gnu++2a -fPIC -MD -MT _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o -MF _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o.d -o _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o -c /tmp/tmp2fz4ptpk/build/_deps/cifpp-src/src/pdb/cif2pdb.cpp
  In file included from /tmp/tmp2fz4ptpk/build/_deps/cifpp-src/include/cif++.hpp:29,
                   from /tmp/tmp2fz4ptpk/build/_deps/cifpp-src/src/pdb/cif2pdb.cpp:27:
  /tmp/tmp2fz4ptpk/build/_deps/cifpp-src/include/cif++/utilities.hpp:193:2: error: ‘requires’ does not name a type
    193 |  requires std::is_assignable_v<std::string_view, T>
        |  ^~~~~~~~
  ninja: build stopped: subcommand failed.

  *** CMake build failed
  error: subprocess-exited-with-error

  × Building wheel for alphafold3 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /data/xiezhihao/software/micromamba/envs/af3/bin/python /data/xiezhihao/software/micromamba/envs/af3/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmppbt09zuj
  cwd: /data/xiezhihao/software/alphafold3
  Building wheel for alphafold3 (pyproject.toml) ... error
  ERROR: Failed building wheel for alphafold3
Failed to build alphafold3
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (alphafold3)
`

I've seen ISSUES [#100 ], but this bug has been fixed. 
I don't know how to solve the problem. Help me.
My Linux distribution is Ubuntu, NVIDIA GeForce RTX 3090, and my cuda 11.4.

**Comment by ZhihaoXie** [comment]
Thank you very much for your comments.
Yes, I know this problem was not caused by the bug of issuse 100. I tried to install alphafold3 (docker) in my environment, but still repeated the problem. Like this:
`#0 601.6       /tmp/tmpdab7iat6/build/_deps/cifpp-src/include/cif++/utilities.hpp:193:2: error: ‘requires’ does not name a type
#0 601.6         193 |  requires std::is_assignable_v<std::string_view, T>
#0 601.6             |  ^~~~~~~~
#0 601.6       [200/235] Building CXX object _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o
#0 601.6       FAILED: _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o
#0 601.6       /usr/bin/x86_64-linux-gnu-g++  -pthread -DBOOST_REGEX_STANDALONE=1 -DCACHE_DIR=\"/tmp/tmpdab7iat6/wheel/platlib/var/cache/libcifpp\" -DDATA_DIR=\"/tmp/tmpdab7iat6/wheel/platlib/share/libcifpp\" -DUSE_BOOST_REGEX=1 -I/tmp/tmpdab7iat6/build/_deps/cifpp-src/include -I/tmp/tmpdab7iat6/build/_deps/boost-rx-src/include -I/tmp/tmpdab7iat6/build/_deps/my-eigen3-src -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -O3 -DNDEBUG -std=gnu++2a -fPIC -MD -MT _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o -MF _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o.d -o _deps/cifpp-build/CMakeFiles/cifpp.dir/src/pdb/cif2pdb.cpp.o -c /tmp/tmpdab7iat6/build/_deps/cifpp-src/src/pdb/cif2pdb.cpp
#0 601.6       In file included from /tmp/tmpdab7iat6/build/_deps/cifpp-src/include/cif++.hpp:29,
#0 601.6                        from /tmp/tmpdab7iat6/build/_deps/cifpp-src/src/pdb/cif2pdb.cpp:27:
#0 601.6       /tmp/tmpdab7iat6/build/_deps/cifpp-src/include/cif++/utilities.hpp:193:2: error: ‘requires’ does not name a type
#0 601.6         193 |  requires std::is_assignable_v<std::string_view, T>
#0 601.6             |  ^~~~~~~~
#0 601.6       ninja: build stopped: subcommand failed.
#0 601.6       
#0 601.6       *** CMake build failed
#0 601.6       [end of output]`
Maybe I'll solve the **libcifpp** problem.
Perhaps you have any good advice. Thank you very much.

**Comment by jsspencer** [comment]
What version of gcc are you using? I suspect you need a newer compiler version -- libcifpp requires C++17

**Comment by ZhihaoXie** [comment]
My version of gcc is 9.4.0. So I need to update my gcc version. I'll give it a try.
Thank you.
![image](https://github.com/user-attachments/assets/32ab26e1-e2e0-4a86-aadd-910253cbf372)


> What version of gcc are you using? I suspect you need a newer compiler version -- libcifpp requires C++17

**Comment by Augustin-Zidek** [comment]
Indeed, GCC 9.4 is quite old, I recommend upgrading to GCC 14. You might get away with GCC 11, which supports C++17, but we might be using some C++20 features as well, so I recommend going for the latest GCC version if possible.

**Comment by ZhihaoXie** [comment]
> Indeed, GCC 9.4 is quite old, I recommend upgrading to GCC 14. You might get away with GCC 11, which supports C++17, but we might be using some C++20 features as well, so I recommend going for the latest GCC version if possible.

OK. I will. Thank you for your advice.
